ERCS/erc-7580.md

```diff
@@ -0,0 +1,92 @@
+category: ERC
+created: 2023-12-13
+---
+
+## Abstract
+
+This proposal aims to establish an inter-dapp tracking standard inferface for dapp interactions and support for more complicated business logics.
+
+As dapps shows their values and more dapps are coming onto chain,there's plenty of requirements for interactions between dapps. For one like, we have two entities holding their respective contracts: contract A and contract B. Then A targets on those users who did some key moves(commit specific functions) in contract B and would like to give bonus/airdrop to these users. Sure B would also get incentives in the meanwhile. To connect all these dots, B needs to identity these users, verify they're coming for the A's bonus. Hence, we need a track mechanism to facilitate such business.
+
+This track mechanism try to offer a neaty and convenient interface for contracts to interact with each other.
+
+## Motivation
+
+Currently, for dapp interaction tracking scenarios, we don't have a shared standard solution. Such standard interface would facilitate the effiency and booms the above use cases, specially in social and game realms. We can expect it thriving later and turn into promotion/ads, a web3 promotions/ads solution.
+
+## Specification
+
+The keywords “MUST,” “MUST NOT,” “REQUIRED,” “SHALL,” “SHALL NOT,” “SHOULD,” “SHOULD NOT,” “RECOMMENDED,” “MAY,” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+**Interface**
+
+This protocol standardizes how to keep tracks on inter dapps operations, which initially offers 2 main methos onTrackStart and onTrackEnd.
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.10;
+
+interface IERCXXX {
+    // Events
+
+    /// @dev Emits when track starts.
+    /// @param track_id track id
+    /// @param contract_address the address of tracking contract
+    /// @param function_hash the hash of tracking function
+    event onTrackStartRecorded(uint256 track_id, address contract_address, bytes32 function_hash);
+
+    /// @dev Emits when track ends.
+    /// @param owner The owner address of the newly minted combo
+    /// @param comboId The newly minted combo identifier
+    event onTrackStartRecorded(uint256 track_id, address contract_address, bytes32 function_hash);
+
+    // Functions
+
+    /// @dev Track a specified contract function start move.
+    /// @param track_id track id
+    /// @param contract_address the address of tracking contract
+    /// @param function_hash the hash of tracking function
+    function onTrackStart(uint256 track_id, address contract_address, bytes32 function_hash) external;
+
+    /// @dev Track a specified contract function end move.
+    /// @param track_id track id
+    /// @param contract_address the address of tracking contract
+    /// @param function_hash the hash of tracking function
+    function onTrackEnd(uint256 track_id, address contract_address, bytes32 function_hash);
+
+    // TODO: may need more
```

Comment:
```suggestion
```

It would be best to omit this type of comment. They're easy to miss during reviews, and look rather silly in Final proposals.

If you'd like, you can leave yourself an HTML-style comment (eg. `<!-- TODO: add more interface functions -->`) outside of the code fence, and the linter will make sure you remove it before leaving Draft.